components/admin/lessons/AdminLessonNav.tsx

```diff
@@ -0,0 +1,72 @@
+import { toUpper } from 'lodash'
+import React, { useState } from 'react'
+import { Tab, Col, Nav } from 'react-bootstrap'
+import styles from '../../../scss/adminLessonNav.module.scss'
+
+type AdminLessonNavProps = {
+  navItems: { value: string }[]
+  tabs: (() => JSX.Element)[]
+}
+
+const validateProps = ({ navItems, tabs }: AdminLessonNavProps) => {
+  if (navItems.length !== tabs.length)
+    throw new Error(
+      `navItems and tabs should have the same count. navItems: ${navItems.length} -- tabs: ${tabs.length}`
+    )
+
+  const mapObjArrayToValues = navItems.map(item => item.value)
+
+  if (new Set(mapObjArrayToValues).size !== mapObjArrayToValues.length)
+    throw new Error('navItems should have unique items value. No duplicates')
```

Comment:
What happens when these validation errors get thrown in the front end?

Reply:
An error will appear in the page indicating that the way the developer used the component is wrong.

Reply:
We should use typescript to make sure that navItems and tab props are the same array length
https://stackoverflow.com/questions/62206320/typescript-require-that-two-arrays-be-the-same-length

Reply:
Not sure if its exactly possible(i haven't tried myself), but please try and see if its possible with typescript instead of

Reply:
I'll look into it